apiclient/container_inspect.go

```diff
@@ -0,0 +1,748 @@
+/*
+Copyright 2019 Thomas Weber
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiclient
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+// ContainerInspect data takes a name or ID of a container returns the inspection data
+func (c *APIClient) ContainerInspect(ctx context.Context, name string) (InspectContainerData, error) {
+
+	var inspectData InspectContainerData
+
+	res, err := c.Get(ctx, fmt.Sprintf("/%s/libpod/containers/%s/json", PODMAN_API_VERSION, name))
+	if err != nil {
+		return inspectData, err
+	}
+
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return inspectData, fmt.Errorf("unknown error, status code: %d", res.StatusCode)
+	}
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return inspectData, err
+	}
+	err = json.Unmarshal(body, &inspectData)
+	if err != nil {
+		return inspectData, err
+	}
+
+	return inspectData, nil
+}
+
+// -------------------------------------------------------------------------------------------------------
```

Comment:
if we have these in their own file we could eventually have automation to ensure they are in-sync, its also a lot of info so putting them in a separate file may help us organize our client code vs podman structs